go/libkb/chain_link.go

```diff
@@ -878,19 +979,23 @@ func (c *ChainLink) VerifySigWithKeyFamily(ckf ComputedKeyFamily) (err error) {
 	return nil
 }
 
-func ImportLinkFromServer(g *GlobalContext, parent *SigChain, jw *jsonw.Wrapper, selfUID keybase1.UID) (ret *ChainLink, err error) {
+func ImportLinkFromServer(g *GlobalContext, parent *SigChain, data []byte, selfUID keybase1.UID) (ret *ChainLink, err error) {
 	var id LinkID
 
-	if ph := jw.AtKey("payload_hash"); !ph.IsNil() {
-		id, err = GetLinkID(ph)
+	if ph, err := jsonparser.GetString(data, "payload_hash"); err == nil {
+		id, err = LinkIDFromHex(ph)
 		if err != nil {
 			return nil, err
 		}
 	}
 	ret = NewChainLink(g, parent, id)
-	if err = ret.Unpack(false, selfUID, jw); err != nil {
+	if err = ret.Unpack(false, selfUID, data); err != nil {
+		g.Log.Debug("Unpack error: %s", err)
 		return nil, err
 	}
+
```

Comment:
did you benchmark without this?

Reply:
The LinkCache line?

Reply:
yes

Reply:
Not recently.  If LinkCache is useful, then seems like we should cache here too, not just when loaded from storage.

But I can try it with it off.